changelog/2024-02-23T16_56_56+01_00_fix2628

```diff
@@ -0,0 +1 @@
+FIXED: Clash no longer errors out in the netlist generation stage out when a polymorphic function is applied to type X in one alternative of a case-statement and applied to a newtype wrapper of type X in a different alternative. See [#2828](https://github.com/clash-lang/clash-compiler/issues/2628)
```

Comment:
Changelog LGTM apart from this little detail:

```suggestion
FIXED: Clash no longer errors out in the netlist generation stage when a polymorphic function is applied to type X in one alternative of a case-statement and applied to a newtype wrapper of type X in a different alternative. See [#2828](https://github.com/clash-lang/clash-compiler/issues/2628)
```

Reply:
Thanks, I'll fix that :)